rules/internal/utils/find.go

```diff
@@ -126,3 +126,26 @@ func GetRepeatedMessageFields(m *desc.MessageDescriptor) []*desc.FieldDescriptor
 
 	return fields
 }
+
+// FindFieldDotNotation attempts to find the field within the given message
+// identified by the dot-notation path e.g. book.name. This is especially useful
+// for resolving path variables in google.api.http and nested
+// fields in google.api.method_signature annotations.
+func FindFieldDotNotation(msg *desc.MessageDescriptor, f string) *desc.FieldDescriptor {
```

Comment:
The f string parameter might be named "path" or something that would help reader understand the meaning.

Reply:
I've already got a variable named `path` in the method body, so I've renamed `f` to `ref`.